server/src/main/java/com/saralein/server/filesystem/FileIO.java

```diff
@@ -1,15 +1,31 @@
 package com.saralein.server.filesystem;
 
 import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
 
 public class FileIO implements IO {
     @Override
-    public byte[] read(Path path) throws IOException {
+    public byte[] readAllBytes(Path path) throws IOException {
         return Files.readAllBytes(path);
     }
 
+    @Override
+    public byte[] readByteRange(Path file, int start, int end) throws IOException {
```

Comment:
You’ve got this nice range abstraction? Why not pass it all the way through to the place it’s needed?

Reply:
OK, I can do that.  At one point I had named that method `partialRead`.  But with a name like `readByteRange`, giving it the range makes total sense.  I think I never reevaluated the parameters.